### PR TITLE
:penguin: Add Jetson agx orin image

### PR DESCRIPTION
**Pull request by mudler**
**What this PR does / why we need it**: Adds initial support for Nvidia AGX Orin. Will follow up and iterate with the image hooked from the CI.

The difference with the other flavors is that instead of a flashable SD image, we prepare the partitions .img files that are ready to be flashed to the eeprom, and we just pack that as part of the releases so others don't have to re-create those images again.

Next:
- Try upgrades
- Create equivalent image in provider-kairos
- Add documentation on how to flash the board with the image files
- See what we can do about resets https://github.com/kairos-io/kairos/issues/622 

Related to #1395

### CODE REVIEW

File: overlay/files-nvidia/etc/cos/bootargs.cfg [hunk 1]
```diff
@@ -0,0 +1,9 @@
+set kernel=/boot/vmlinuz
+
+if [ -n "$recoverylabel" ]; then
+    set kernelcmd="console=tty1 console=ttyTCU0,115200 root=live:LABEL=$recoverylabel net.ifnames=1 rd.live.dir=/ rd.live.squashimg=$img panic=5 rd.cos.oemtimeout=10"
```
Comment (Itxaka) [inline overlay/files-nvidia/etc/cos/bootargs.cfg:4]: we need a template for this file asap.

File: Earthfile [hunk 2]
```diff
@@ -477,6 +491,39 @@ arm-image:
   END
   SAVE ARTIFACT /build/$IMAGE_NAME.sha256 img-sha256 AS LOCAL build/$IMAGE_NAME.sha256
 
+prepare-arm-image:
```
Comment (Itxaka) [inline Earthfile:494]: cant we reuse the existing arm image stuff? What is the difference from other arm images?

File: Earthfile [hunk 3]
```diff
@@ -55,7 +55,17 @@ all-arm:
       BUILD --platform=linux/arm64 +trivy-scan --MODEL=rpi64
       BUILD --platform=linux/arm64 +grype-scan --MODEL=rpi64
   END
-  BUILD +arm-image --MODEL=rpi64
+  
+  IF [[ "$FLAVOR" = "ubuntu-20-lts-arm-nvidia-jetson-agx-orin" ]]
```
Comment (Itxaka) [inline Earthfile:59]: do we expect other flavors for this boards or ubuntu would be the only one?

File: .github/workflows/image-arm.yaml [hunk 4]
```diff
@@ -65,12 +65,19 @@ jobs:
...
+        with:
+          repository: quay.io/kairos/packages
+          packages: utils/earthly
       - name: Build  🔧
         env:
           FLAVOR: ${{ matrix.flavor }}
           MODEL: ${{ matrix.model }}
         run: |
-          ./earthly.sh +all-arm --IMAGE_NAME=kairos-$FLAVOR-latest.img --IMAGE=quay.io/kairos/core-$FLAVOR:latest --MODEL=$MODEL --FLAVOR=$FLAVOR
+          earthly config "global.conversion_parallelism" "1"
```
Comment (Itxaka) [inline .github/workflows/image-arm.yaml:78]: comment with reason?

**Review by Itxaka — DISMISSED** [review]

File: Earthfile (hunk 2 above)
Comment (mudler) [reply Earthfile:494]: we don't actually create an SD card here, we just prepare the partitions

File: Earthfile (hunk 3 above)
Comment (mudler) [reply Earthfile:59]: can't tell, probably this one will be the only one for a long time.. until nvidia releases another version or we have some reques

File: .github/workflows/image-arm.yaml (hunk 4 above)
Comment (mudler) [reply .github/workflows/image-arm.yaml:78]: dunno if it works yet, debugging as it fails in transferring generated assets...

**Review by mauromorales — DISMISSED** [review]
:clap:

**Comment by mudler** [comment]
nice :D 
```
failed to register layer: ApplyLayer exit status 1 stdout:  stderr: write /usr/src/linux-headers-5.10.104-tegra-ubuntu20.04_aarch64/kernel-5.10/scripts/recordmcount.pl: no space left on device: exit status 1
```

**Comment by mudler** [comment]
GREEEN!

**Pull request merged**